docs/examples/kube2iam.rst

```diff
@@ -0,0 +1,246 @@
+    provider "aws" {
+        region = "${var.aws_region}"
+    }
+
+    variable "aws_region" {
+        description = "AWS Region you want to deploy it in"
+        default     = "eu-west-1"
+    }
+
+    variable "cluster_name" {
+        description = "Name of the cluster"
+    }
+
+    variable "instance_iam_role_arn" {
+        description = "ARN of the instance IAM role"
+    }
+
+    resource "aws_iam_role" "test_role" {
+        name = "test_role"
+        path = "/kube2iam_${var.cluster_name}/"
+
+        assume_role_policy = "${data.aws_iam_policy_document.test_role.json}"
+    }
+
+    data "aws_iam_policy_document" "test_role" {
+        statement {
+            sid = "1"
+
+            actions = [
+                "sts:AssumeRole",
+            ]
+
+            principals {
+                type        = "AWS"
+                identifiers = ["${var.instance_iam_role_arn}"]
+            }
+        }
+    }
+
+    resource "aws_iam_role_policy" "test_role_policy" {
+        name = "test_policy"
+        role = "${aws_iam_role.test_role.id}"
+
+        policy = "${data.aws_iam_policy_document.test_role_policy.json}"
+    }
+
+    data "aws_iam_policy_document" "test_role_policy" {
+        statement {
+            sid = "1"
+
+            actions = [
+                "s3:ListBucket",
+            ]
+
+            resources = [
+                "*",
+            ]
+        }
+    }
+
+    output "test_role" {
+        value = "${aws_iam_role.test_role.arn}"
+    }
+
+Now you can run this Terraform project the following way:
+
+.. code-block:: bash
+
+    terraform init
+    terraform apply -var cluster_name=example -var region=eu-west-1 -var instance_iam_role_arn=arn:aws:iam::xxxxxxx:role/my-instance-role
+
+
+When you create a role, you need to make sure you deploy it in the correct
+``path`` and also add an assume role policy to it. That assume role policy
+needs to grant access to the role ARN that is attached to the instances.
+In our example Terraform project above we solved that by adding a variable for
+the ``instance_iam_role_arn`` and the ``cluster_name``.
+
+.. warning::
+   Make sure you use the IAM role ARN attached to your worker Kubernetes 
+   instances as input for ``instance_iam_role_arn``. You can retrieve the
+   IAM role ARN through the AWS console.
+   Don't confuse this with the earlier created IAM policy. 
+
+With the output of the test role, you can add that as an annotation to your deployment.
+
+.. code-block:: yaml
+
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+        name: nginx-deployment
+    spec:
+        replicas: 3
+        template:
+            metadata:
+                annotations:
+                    iam.amazonaws.com/role: role-arn
+                labels:
+                    app: nginx
+            spec:
+                containers:
+                - name: nginx
+                  image: nginx:1.9.1
+                  ports:
+                  - containerPort: 80
```

Comment:
should change this to something that actually uses to AWS APIs, e.g the following from https://github.com/jtblin/kube2iam#kubernetes-annotation:
```
apiVersion: v1
kind: Pod
metadata:
  name: aws-cli
  labels:
    name: aws-cli
  annotations:
    iam.amazonaws.com/role: role-arn
spec:
  containers:
  - image: fstab/aws-cli
    command:
      - "/home/aws/aws/env/bin/aws"
      - "s3"
      - "ls"
      - "some-bucket"
    name: aws-cli
```